crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
Yes. My Ruby-fu is low but if I do not misinterpret the code, you are only attending the first batch of threads. You need to iterate until the queue (that if from the same size that the `locks`) is indeed empty.

Or you are iterating in some other place?